src/wrappers/themis/react-native-themis/ios/RCTThemis.m

```diff
@@ -50,131 +50,154 @@ + (BOOL)requiresMainQueueSetup
+            break;
+    }
 
-  NSArray  *privateKey = [RCTThemis dataSerialize: keypair.privateKey];
-  NSArray   *publicKey = [RCTThemis dataSerialize: keypair.publicKey];
-  
-  NSDictionary *dictionary = @{
-       @"private" : privateKey,
-       @"public"  : publicKey
-  };
-  callback(@[dictionary]);
+    NSArray  *privateKey = [RCTThemis dataSerialize: keypair.privateKey];
+    NSArray   *publicKey = [RCTThemis dataSerialize: keypair.publicKey];
+
+    NSDictionary *dictionary = @{
+        @"private" : privateKey,
+        @"public"  : publicKey
+    };
+    successCallback(@[dictionary]);
 }
 
 
 RCT_EXPORT_METHOD(symmetricKey:(RCTResponseSenderBlock)callback)
 {
-  NSData *symmetricKey = TSGenerateSymmetricKey();
+    NSData *symmetricKey = TSGenerateSymmetricKey();
     NSArray *masterKey = [RCTThemis dataSerialize: symmetricKey];
-  callback(@[masterKey]);
+    callback(@[masterKey]);
 }
 
 - (TSCellSeal *)newSealMode: (NSArray*) symmetricKey
 {
-  @try {
-    NSData *masterKey = [RCTThemis dataDeserialize: symmetricKey];
-    TSCellSeal *cell  = [[TSCellSeal alloc] initWithKey:masterKey];
-    return cell;
-  }
-  @catch (NSException *e) {
-    @throw e; // rethrow to catch in final function
-  }
-
+    @try {
+        NSData *masterKey = [RCTThemis dataDeserialize: symmetricKey];
+        TSCellSeal *cell  = [[TSCellSeal alloc] initWithKey:masterKey];
+        return cell;
+    }
+    @catch (NSException *e) {
+        @throw e; // rethrow to catch in final function
+    }
 }
 
 
-
-
-
 RCT_EXPORT_METHOD(secureCellSealWithSymmetricKeyEncrypt:(NSArray*) symmetricKey
                   plaintext: (NSString*)plaintext
                   context: (NSString*)context
                   successCallback: (RCTResponseSenderBlock)successCallback
                   errorCallback: (RCTResponseErrorBlock)errorCallback)
 {
 
-  TSCellSeal *cell;
-  @try {
-    cell  = [self newSealMode:symmetricKey];
-  }
-  @catch (NSException *e) {
-    NSError* error = SCERROR(BYTEOVERFLOW, e.reason);
-    errorCallback(error);
-    return;
-  }
-  
-  NSData *txt  = [plaintext dataUsingEncoding:NSUTF8StringEncoding];
-  NSData *ctx  = [context dataUsingEncoding:NSUTF8StringEncoding];
-  NSData *encrypted = [cell encrypt:txt context:ctx];
-  NSArray *result = [RCTThemis dataSerialize:encrypted];
-  successCallback(@[result]);
+    TSCellSeal *cell;
+    @try {
+        cell  = [self newSealMode:symmetricKey];
+    }
+    @catch (NSException *e) {
+        NSNumber* errorCode = e.userInfo[@"errorCode"]; // Exception possible from deserialize
+        NSError* error = SCERROR([errorCode intValue], e.reason);
+        errorCallback(error);
+        return;
+    }
+
+    if (cell == nil) {
+        // It is possible for few reasons
+        NSError* error = SCERROR(CELL_SEAL_NIL_ERROR, @CELL_SEAL_RETURNED_NIL);
+        errorCallback(error);
+        return;
+    }
+
+    NSData *txt  = [plaintext dataUsingEncoding:NSUTF8StringEncoding];
```

Comment:
there is too easy to mistype variables and write `ctx` instead of `txt`. Can we use more verbose variable names? For example `plaintext` and `context`?

Reply:
Would you like plaintextBinary and contextBinary names? Because context and plaintext are already busy in that functions are declared as parameters. So, I decided to use 'ctx' and 'txt' names to indicate that the variables used for binary data.

Reply:
> Would you like plaintextBinary and contextBinary names?

looks ok. much better than txt/ctx where easy to mistype